test/cmd/discovery.sh

```diff
@@ -87,10 +87,12 @@ run_resource_aliasing_tests() {
 
   # all 4 cassandra's might not be in the request immediately...
   # first option with :: suffix is for possible service.kubernetes.io/headless
-  # label with "" value
-  kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:cassandra::' || \
-  kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:cassandra:' || \
-  kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:' || \
+  # label with "" value.
+  # The output of the first three assertions are redirected to /dev/null so that error messages are not displayed when
+  # a subsequent assertion may pass. If it makes it to the final assertion, and still fails, the output will be displayed
+  kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:cassandra::' > /dev/null || \
```

Comment:
I see, but the error is still a little misleading, anything we can do to improve it?

```
kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:cassandra::' > /dev/null || \
kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:cassandra:' > /dev/null || \
kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:cassandra:' > /dev/null || \
kube::test::get_object_assert "$object" "$request" 'cassandra:cassandra:' >/dev/null || {
  # You can almost certainly improve that
  echo "Couldn't find any of 'cassandra:cassandra:cassandra:cassandra::', 'cassandra:cassandra:cassandra:cassandra:' ..."
  false
}

Reply:
Sure, let me take a look at it and see if I can make it more descriptive...

Reply:
I would rather suggest that you possibly make this into a change to `hack/lib/test.sh` so that others can re-use, and it will probably be less confusing to read.

Reply:
Makes sense. I was trying to keep the change small, but I think it would be good to have a general function to write out the error. I will refactor this into a general function in test.sh, thanks!